lib/hls/hls_parser.js

```diff
@@ -2236,8 +2248,11 @@ shaka.hls.HlsParser = class {
     for (const tag of tags) {
       goog.asserts.assert(tag.name == 'EXT-X-MEDIA',
           'Should only be called on media tags!');
+      const getFakeUri = () => {
+        return shaka.hls.HlsParser.FAKE_MUXED_URL_ + this.globalId_++;
+      };
```

Comment:
Is this counter really needed? It creates link with HLS parser, which might be problematic for GC. If really needed, I'd suggest dereferencing:
```suggestion
      const id = ++this.globalId_;
      const getFakeUri = () => {
        return shaka.hls.HlsParser.FAKE_MUXED_URL_ + id;
      };
```